indigo_app/forms.py

```diff
@@ -250,6 +251,56 @@ def filter_queryset(self, queryset):
         return queryset
 
 
+class WorkFilterForm(forms.Form):
+    search_string = forms.CharField()
+    stub = forms.ChoiceField(choices=[('excl', 'excl'), ('all', 'all'), ('only', 'only')])
+    status = forms.MultipleChoiceField(choices=[('published', 'published'), ('draft', 'draft')])
+    subtype = forms.ChoiceField(choices=[('-', '-'), ('by-law', 'by-law'), ('gn', 'gn'), ('p', 'p'), ('si', 'si')])
+    sortby = forms.ChoiceField(choices=[('-updated_at', '-updated_at'), ('updated_at', 'updated_at'), ('title', 'title'), ('-title', '-title'), ('frbr_uri', 'frbr_uri')])
+
+    def __init__(self, country, *args, **kwargs):
+        self.country = country
+        super(WorkFilterForm, self).__init__(*args, **kwargs)
+
+    def filter_queryset(self, queryset):
+        print(self.cleaned_data)
+        if self.cleaned_data.get('search_string'):
+            queryset = queryset.filter(Q(title__icontains=self.cleaned_data['search_string']) | Q(frbr_uri__icontains=self.cleaned_data['search_string']))
+
+        if self.cleaned_data.get('stub'):
+            if self.cleaned_data['stub'] == 'excl':
+                queryset = queryset.filter(stub=False)
+
+            elif self.cleaned_data['stub'] == 'only':
+                queryset = queryset.filter(stub=True)
+
+        # TODO: revisit this
+        if self.cleaned_data.get('status'):
+            if self.cleaned_data['status'] == ['draft']:
+                queryset = queryset.filter(document__draft=True)            
+            elif self.cleaned_data['status'] == ['published']:
+                queryset = queryset.filter(document__draft=False)
+
+        if self.cleaned_data.get('sortby'):
+            queryset = queryset.order_by(self.cleaned_data.get('sortby'))        
+        
+        # filter by subtype indicated on frbr_uri
+        if self.cleaned_data.get('subtype'):
+            if self.cleaned_data['subtype'] == 'by-law':
```

Comment:
I suggest the below. We trust the value in cleaned_data, and adding the `/act/` will ensure it's matching in the right place.

```python
queryset = queryset.filter(frbr_uri__contains='/act/%s/' % self.cleaned_data['subtype'])
```